src/openrct2/actions/TrackRemoveAction.cpp

```diff
@@ -421,7 +421,7 @@ GameActions::Result TrackRemoveAction::Execute() const
         // If the removed tile is a station modify station properties.
         // Don't do this if the ride is simulating and the tile is a ghost to prevent desyncs.
         if (entranceDirections & TRACK_SEQUENCE_FLAG_ORIGIN
-            && !(ride->status == RideStatus::Simulating && tileElement->Flags & TILE_ELEMENT_FLAG_GHOST)
+            && (!(GetFlags() & GAME_COMMAND_FLAG_GHOST) || GetFlags() & GAME_COMMAND_FLAG_TRACK_DESIGN)
```

Comment:
This looks a bit strange, wrap the and operation in parenthesis to make this more clear, ideally we should have a getter HasFlags all along.